# Air quality analysis in Barcelona

## 1. Data import and cleaning

In [1]:
## import libraries

import pandas as pd 
import sqlalchemy 
import datetime 

In [2]:
raw_data = pd.read_csv('air_quality_Nov2017.csv')

In [3]:
raw_data.head()

Station Air Quality  Longitude  Latitude O3 Hour  \
0        Barcelona - Sants        Good     2.1331   41.3788     NaN   
1     Barcelona - Eixample    Moderate     2.1538   41.3853      0h   
2       Barcelona - Gràcia        Good     2.1534   41.3987      0h   
3   Barcelona - Ciutadella        Good     2.1874   41.3864      0h   
4  Barcelona - Vall Hebron        Good     2.1480   41.4261      0h   

  O3 Quality  O3 Value NO2 Hour NO2 Quality  NO2 Value PM10 Hour PM10 Quality  \
0        NaN       NaN       0h        Good       84.0       NaN          NaN   
1       Good       1.0       0h    Moderate      113.0        0h         Good   
2       Good      10.0       0h        Good       73.0       NaN          NaN   
3       Good       2.0       0h        Good       86.0       NaN          NaN   
4       Good       7.0       0h        Good       69.0       NaN          NaN   

   PM10 Value        Generated   Date Time  
0         NaN  01/11/2018 0:00  1541027104  
1        36.0  01/11/2018 0:00  1541027104  
2         NaN  01/11/2018 0:00  1541027104  
3         NaN  01/11/2018 0:00  1541027104  
4         NaN  01/11/2018 0:00  1541027104

We remove the columns that are not needed for our analysis. In this case, the columns that are needed are: 
- Station
- Air Quality 
- O3 Quality
- 03 Value
- NO2 Quality
- NO2 Value
- PM10 Quality
- PM10 Value 
- Generated 
- Date Time 

In [4]:
print(raw_data.columns.tolist())

['Station', 'Air Quality', 'Longitude', 'Latitude', 'O3 Hour', 'O3 Quality', 'O3 Value', 'NO2 Hour', 'NO2 Quality', 'NO2 Value', 'PM10 Hour', 'PM10 Quality', 'PM10 Value', 'Generated', 'Date Time']


In [5]:
raw_data_selected = raw_data.copy()

raw_data_selected = raw_data_selected[['Station', 'Air Quality', 'O3 Quality', 'O3 Value', 'NO2 Quality', 'NO2 Value', 'PM10 Quality', 'PM10 Value', 'Generated']]

raw_data_selected.head()

Station Air Quality O3 Quality  O3 Value NO2 Quality  \
0        Barcelona - Sants        Good        NaN       NaN        Good   
1     Barcelona - Eixample    Moderate       Good       1.0    Moderate   
2       Barcelona - Gràcia        Good       Good      10.0        Good   
3   Barcelona - Ciutadella        Good       Good       2.0        Good   
4  Barcelona - Vall Hebron        Good       Good       7.0        Good   

   NO2 Value PM10 Quality  PM10 Value        Generated  
0       84.0          NaN         NaN  01/11/2018 0:00  
1      113.0         Good        36.0  01/11/2018 0:00  
2       73.0          NaN         NaN  01/11/2018 0:00  
3       86.0          NaN         NaN  01/11/2018 0:00  
4       69.0          NaN         NaN  01/11/2018 0:00

### Analyzing the type of each attribute of the dataset

In [6]:

raw_data_selected.dtypes

Station          object
Air Quality      object
O3 Quality       object
O3 Value        float64
NO2 Quality      object
NO2 Value       float64
PM10 Quality     object
PM10 Value      float64
Generated        object
dtype: object

### Changing type of date to DateTime

In [7]:
raw_data_selected['date_corrected'] = pd.to_datetime(raw_data_selected["Generated"])

raw_data_selected.set_index("date_corrected", inplace=True) # Inplace reassign dataset to indexed dataset. Is as if dataset_NEW=dataset_OLD

raw_data_selected = raw_data_selected.drop(["Generated"], axis=1)

In [16]:
raw_data_selected.index

DatetimeIndex(['2018-01-11 00:00:00', '2018-01-11 00:00:00',
               '2018-01-11 00:00:00', '2018-01-11 00:00:00',
               '2018-01-11 00:00:00', '2018-01-11 00:00:00',
               '2018-01-11 00:00:00', '2018-01-11 00:00:00',
               '2018-01-11 01:00:00', '2018-01-11 01:00:00',
               ...
               '2018-11-30 22:00:00', '2018-11-30 22:00:00',
               '2018-11-30 23:00:00', '2018-11-30 23:00:00',
               '2018-11-30 23:00:00', '2018-11-30 23:00:00',
               '2018-11-30 23:00:00', '2018-11-30 23:00:00',
               '2018-11-30 23:00:00', '2018-11-30 23:00:00'],
              dtype='datetime64[ns]', name='date_corrected', length=5744, freq=None)

In [19]:
raw_data_selected

Station Air Quality O3 Quality  \
date_corrected                                                         
2018-01-11 00:00:00         Barcelona - Sants        Good        NaN   
2018-01-11 00:00:00      Barcelona - Eixample    Moderate       Good   
2018-01-11 00:00:00        Barcelona - Gràcia        Good       Good   
2018-01-11 00:00:00    Barcelona - Ciutadella        Good       Good   
2018-01-11 00:00:00   Barcelona - Vall Hebron        Good       Good   
2018-01-11 00:00:00   Barcelona - Palau Reial        Good       Good   
2018-01-11 00:00:00      Barcelona - Poblenou        Good        NaN   
2018-01-11 00:00:00  Barcelona - Observ Fabra        Good       Good   
2018-01-11 01:00:00         Barcelona - Sants        Good        NaN   
2018-01-11 01:00:00      Barcelona - Eixample        Good       Good   
2018-01-11 01:00:00        Barcelona - Gràcia        Good       Good   
2018-01-11 01:00:00    Barcelona - Ciutadella        Good       Good   
2018-01-11 01:00:00   Barcelona - Vall Hebron        Good       Good   
2018-01-11 01:00:00   Barcelona - Palau Reial        Good       Good   
2018-01-11 01:00:00      Barcelona - Poblenou        Good        NaN   
2018-01-11 01:00:00  Barcelona - Observ Fabra        Good       Good   
2018-01-11 02:00:00         Barcelona - Sants        Good        NaN   
2018-01-11 02:00:00      Barcelona - Eixample        Good       Good   
2018-01-11 02:00:00        Barcelona - Gràcia        Good       Good   
2018-01-11 02:00:00    Barcelona - Ciutadella        Good       Good   
2018-01-11 02:00:00   Barcelona - Vall Hebron        Good       Good   
2018-01-11 02:00:00   Barcelona - Palau Reial        Good       Good   
2018-01-11 02:00:00      Barcelona - Poblenou        Good        NaN   
2018-01-11 02:00:00  Barcelona - Observ Fabra        Good       Good   
2018-01-11 03:00:00         Barcelona - Sants        Good        NaN   
2018-01-11 03:00:00      Barcelona - Eixample        Good       Good   
2018-01-11 03:00:00        Barcelona - Gràcia        Good       Good   
2018-01-11 03:00:00    Barcelona - Ciutadella        Good       Good   
2018-01-11 03:00:00   Barcelona - Vall Hebron        Good       Good   
2018-01-11 03:00:00   Barcelona - Palau Reial        Good       Good   
...                                       ...         ...        ...   
2018-11-30 20:00:00        Barcelona - Gràcia        Good       Good   
2018-11-30 20:00:00    Barcelona - Ciutadella        Good       Good   
2018-11-30 20:00:00   Barcelona - Vall Hebron        Good       Good   
2018-11-30 20:00:00   Barcelona - Palau Reial        Good       Good   
2018-11-30 20:00:00      Barcelona - Poblenou        Good        NaN   
2018-11-30 20:00:00  Barcelona - Observ Fabra        Good       Good   
2018-11-30 21:00:00         Barcelona - Sants        Good        NaN   
2018-11-30 21:00:00      Barcelona - Eixample    Moderate       Good   
2018-11-30 21:00:00        Barcelona - Gràcia        Good       Good   
2018-11-30 21:00:00    Barcelona - Ciutadella        Good       Good   
2018-11-30 21:00:00   Barcelona - Vall Hebron        Good       Good   
2018-11-30 21:00:00   Barcelona - Palau Reial        Good       Good   
2018-11-30 21:00:00      Barcelona - Poblenou        Good        NaN   
2018-11-30 21:00:00  Barcelona - Observ Fabra        Good       Good   
2018-11-30 22:00:00         Barcelona - Sants        Good        NaN   
2018-11-30 22:00:00      Barcelona - Eixample    Moderate         --   
2018-11-30 22:00:00        Barcelona - Gràcia        Good       Good   
2018-11-30 22:00:00    Barcelona - Ciutadella        Good       Good   
2018-11-30 22:00:00   Barcelona - Vall Hebron        Good       Good   
2018-11-30 22:00:00   Barcelona - Palau Reial        Good       Good   
2018-11-30 22:00:00      Barcelona - Poblenou        Good        NaN   
2018-11-30 22:00:00  Barcelona - Observ Fabra        Good       Good   
2018-11-30 23:00:00         Barcelona - Sants        Good        NaN   
2018-1

### General description in Barcelona

In [20]:
raw_data_selected.describe()

O3 Value    NO2 Value   PM10 Value
count  4101.000000  5460.000000  3647.000000
mean     34.082907    35.740293    16.590074
std      22.960687    22.357262     8.065424
min       1.000000     1.000000     2.000000
25%      14.000000    17.000000    10.000000
50%      34.000000    33.000000    15.000000
75%      52.000000    52.000000    22.000000
max     100.000000   117.000000    44.000000

From this table we can say that we have missing values in the general table. However, a general description of Barcelona can be done. The average values for November 2018 in Barcelona are: 
- O3: 34 ug/cm3
- NO2: 35.74 ug/cm3
- PM10: 16.59 ug/cm3

In average terms, Barcelona had a very low qualitative name for O3, very low for NO2 and low for PM10. But, what is the situation in Barcelona? 

### Where has the maximum value of pollution happened in Barcelona? 

In [28]:
# For O3

raw_data_selected[raw_data_selected['O3 Value']==raw_data_selected['O3 Value'].max()]

Station Air Quality O3 Quality  \
date_corrected                                                         
2018-11-29 22:00:00  Barcelona - Observ Fabra        Good       Good   

                     O3 Value NO2 Quality  NO2 Value PM10 Quality  PM10 Value  
date_corrected                                                                 
2018-11-29 22:00:00     100.0        Good        4.0         Good        13.0

In [29]:
# For NO2

raw_data_selected[raw_data_selected['NO2 Value']==raw_data_selected['NO2 Value'].max()]

Station Air Quality O3 Quality  O3 Value  \
date_corrected                                                             
2018-11-22 11:00:00  Barcelona - Gràcia    Moderate       Good       3.0   

                    NO2 Quality  NO2 Value PM10 Quality  PM10 Value  
date_corrected                                                       
2018-11-22 11:00:00    Moderate      117.0         Good        12.0

In [31]:
# For PM10
raw_data_selected[raw_data_selected['NO2 Value']==raw_data_selected['NO2 Value'].max()]

Station Air Quality O3 Quality  O3 Value  \
date_corrected                                                             
2018-11-22 11:00:00  Barcelona - Gràcia    Moderate       Good       3.0   

                    NO2 Quality  NO2 Value PM10 Quality  PM10 Value  
date_corrected                                                       
2018-11-22 11:00:00    Moderate      117.0         Good        12.0

### Missing values

There are several ways to deal with missing values, in this case NaN values.
First of all, we need to know how many NaN values we have in our dataset. 

In [9]:
raw_data_selected.isna().sum()

Station            0
Air Quality        0
O3 Quality      1476
O3 Value        1643
NO2 Quality       55
NO2 Value        284
PM10 Quality    2022
PM10 Value      2097
dtype: int64

#### Method 1: dropping all the NaN values of the entire dataset. 

In this case we have removed all the rows that have a "NaN" value there. If we perform an analysis of what we have done, we can see the total amount of data we have lost 

In [10]:
before_rows = raw_data_selected.shape[0]
print(f"Number of rows before dropping all the NaN values: {before_rows}")

Number of rows before dropping all the NaN values: 5744


In [11]:
clean_air_quality =  raw_data_selected.copy()
clean_air_quality = raw_data_selected.dropna()
clean_air_quality.head()

Station Air Quality O3 Quality  \
date_corrected                                                         
2018-01-11 00:00:00      Barcelona - Eixample    Moderate       Good   
2018-01-11 00:00:00   Barcelona - Palau Reial        Good       Good   
2018-01-11 00:00:00  Barcelona - Observ Fabra        Good       Good   
2018-01-11 01:00:00      Barcelona - Eixample        Good       Good   
2018-01-11 01:00:00   Barcelona - Palau Reial        Good       Good   

                     O3 Value NO2 Quality  NO2 Value PM10 Quality  PM10 Value  
date_corrected                                                                 
2018-01-11 00:00:00       1.0    Moderate      113.0         Good        36.0  
2018-01-11 00:00:00      11.0        Good       57.0         Good        23.0  
2018-01-11 00:00:00      58.0        Good        3.0         Good        25.0  
2018-01-11 01:00:00       6.0        Good       80.0         Good        35.0  
2018-01-11 01:00:00      27.0        Good       38.0         Good        24.0

In [12]:
after_rows = clean_air_quality.shape[0]
print (f"Number of rows after dropping all the NaN values: {after_rows}")

Number of rows after dropping all the NaN values: 2853


In [13]:
data_lost_perc = ((before_rows - after_rows)/before_rows)*100
print(f"The percentage of data that is NaN and has been removed is {data_lost_perc}")


The percentage of data that is NaN and has been removed is 50.330779944289695


If we do it like this, we remove half of the data. 

#### Method 2: Instead of NaN removal, values interpolation 

#### Method 3: Instead of NaN removal, choose the last value of each station 

## Database labeling 

In [55]:
labels = {"Barcelona - Ciutadella":1, "Barcelona - Eixample":2, "Barcelona - Gràcia":3, "Barcelona - Palau Reial":4, 
         "Barcelona - Poblenou": 5, "Barcelona - Sants": 6, "Barcelona - Vall Hebron": 7, 
         "Barcelona - Observ Fabra":8  }

In [38]:
raw_data_selected.head()

Station Air Quality O3 Quality  O3 Value  \
date_corrected                                                             
2018-01-11            Barcelona - Sants        Good        NaN       NaN   
2018-01-11         Barcelona - Eixample    Moderate       Good       1.0   
2018-01-11           Barcelona - Gràcia        Good       Good      10.0   
2018-01-11       Barcelona - Ciutadella        Good       Good       2.0   
2018-01-11      Barcelona - Vall Hebron        Good       Good       7.0   

               NO2 Quality  NO2 Value PM10 Quality  PM10 Value  
date_corrected                                                  
2018-01-11            Good       84.0          NaN         NaN  
2018-01-11        Moderate      113.0         Good        36.0  
2018-01-11            Good       73.0          NaN         NaN  
2018-01-11            Good       86.0          NaN         NaN  
2018-01-11            Good       69.0          NaN         NaN

In [64]:
station_ID=[labels[i] for i in raw_data_selected["Station"]]

raw_data_selected["Station_ID"] = station_ID

raw_data_selected.head()

Station Air Quality O3 Quality  O3 Value  \
date_corrected                                                             
2018-01-11            Barcelona - Sants        Good        NaN       NaN   
2018-01-11         Barcelona - Eixample    Moderate       Good       1.0   
2018-01-11           Barcelona - Gràcia        Good       Good      10.0   
2018-01-11       Barcelona - Ciutadella        Good       Good       2.0   
2018-01-11      Barcelona - Vall Hebron        Good       Good       7.0   

               NO2 Quality  NO2 Value PM10 Quality  PM10 Value  Station_ID  
date_corrected                                                              
2018-01-11            Good       84.0          NaN         NaN           6  
2018-01-11        Moderate      113.0         Good        36.0           2  
2018-01-11            Good       73.0          NaN         NaN           3  
2018-01-11            Good       86.0          NaN         NaN           1  
2018-01-11            Good       69.0          NaN         NaN           7

In [65]:
raw_data_selected = raw_data_selected.drop("Station", axis = 1)

In [68]:
raw_data_selected.head()

Air Quality O3 Quality  O3 Value NO2 Quality  NO2 Value  \
date_corrected                                                           
2018-01-11            Good        NaN       NaN        Good       84.0   
2018-01-11        Moderate       Good       1.0    Moderate      113.0   
2018-01-11            Good       Good      10.0        Good       73.0   
2018-01-11            Good       Good       2.0        Good       86.0   
2018-01-11            Good       Good       7.0        Good       69.0   

               PM10 Quality  PM10 Value  Station_ID  
date_corrected                                       
2018-01-11              NaN         NaN           6  
2018-01-11             Good        36.0           2  
2018-01-11              NaN         NaN           3  
2018-01-11              NaN         NaN           1  
2018-01-11              NaN         NaN           7

In [69]:
news_columns = ["Station_ID", 'Air Quality', 'O3 Quality', 'O3 Value', 'NO2 Quality', 'NO2 Value', 'PM10 Quality', 'PM10 Value']

In [83]:
Barcelona_air_quality = raw_data_selected.reindex(columns=news_columns)

Barcelona_air_quality.head()

Station_ID Air Quality O3 Quality  O3 Value NO2 Quality  \
date_corrected                                                            
2018-01-11               6        Good        NaN       NaN        Good   
2018-01-11               2    Moderate       Good       1.0    Moderate   
2018-01-11               3        Good       Good      10.0        Good   
2018-01-11               1        Good       Good       2.0        Good   
2018-01-11               7        Good       Good       7.0        Good   

                NO2 Value PM10 Quality  PM10 Value  
date_corrected                                      
2018-01-11           84.0          NaN         NaN  
2018-01-11          113.0         Good        36.0  
2018-01-11           73.0          NaN         NaN  
2018-01-11           86.0          NaN         NaN  
2018-01-11           69.0          NaN         NaN

### Secondary table for Station ID

In [77]:
station_info = pd.read_csv('air_stations_Nov2017.csv')
station_info

Station  Longitude  Latitude  \
0   Barcelona - Ciutadella     2.1874   41.3864   
1     Barcelona - Eixample     2.1538   41.3853   
2       Barcelona - Gràcia     2.1534   41.3987   
3  Barcelona - Palau Reial     2.1151   41.3875   
4     Barcelona - Poblenou     2.2045   41.4039   
5        Barcelona - Sants     2.1331   41.3788   
6  Barcelona - Vall Hebron     2.1480   41.4261   
7       Barcelona – Observ     2.1211   41.4176   

                                           Ubication        District Name  \
0                              Parc de la Ciutadella         Ciutat Vella   
1                         Av. Roma - c/ Comte Urgell             Eixample   
2  Plaça Gal·la Placídia (Via Augusta - Travesser...               Gracia   
3        c/ John Maynard Keynes - c/ de Jordi Girona            Les Corts   
4        Plaça Josep Trueta (Pujades - Lope de Vega)           Sant Marti   
5  Jardins de Can Mantega (Joan Güell - Violant d...       Sants-Montjuic   
6  Parc de la Vall d’Hebron - c/ Martí Codolar - ...       Horta-Guinardo   
7                         Ctra Observatori Fabra, 27  Sarrià-Sant Gervasi   

                       Neighborhood Name  
0  Sant Pere, Santa Caterina i la Ribera  
1         la Nova Esquerra de l'Eixample  
2                      la Vila de Gracia  
3                              Pedralbes  
4                            el Poblenou  
5                                  Sants  
6                       la Vall d'Hebron  
7  Vallvidrera, el Tibidabo i les Planes

In [79]:
station_ID_num= [1,2,3,4,5,6,7,8]

station_info['station_ID']= station_ID_num
station_info

Station  Longitude  Latitude  \
0   Barcelona - Ciutadella     2.1874   41.3864   
1     Barcelona - Eixample     2.1538   41.3853   
2       Barcelona - Gràcia     2.1534   41.3987   
3  Barcelona - Palau Reial     2.1151   41.3875   
4     Barcelona - Poblenou     2.2045   41.4039   
5        Barcelona - Sants     2.1331   41.3788   
6  Barcelona - Vall Hebron     2.1480   41.4261   
7       Barcelona – Observ     2.1211   41.4176   

                                           Ubication        District Name  \
0                              Parc de la Ciutadella         Ciutat Vella   
1                         Av. Roma - c/ Comte Urgell             Eixample   
2  Plaça Gal·la Placídia (Via Augusta - Travesser...               Gracia   
3        c/ John Maynard Keynes - c/ de Jordi Girona            Les Corts   
4        Plaça Josep Trueta (Pujades - Lope de Vega)           Sant Marti   
5  Jardins de Can Mantega (Joan Güell - Violant d...       Sants-Montjuic   
6  Parc de la Vall d’Hebron - c/ Martí Codolar - ...       Horta-Guinardo   
7                         Ctra Observatori Fabra, 27  Sarrià-Sant Gervasi   

                       Neighborhood Name  station_ID  
0  Sant Pere, Santa Caterina i la Ribera           1  
1         la Nova Esquerra de l'Eixample           2  
2                      la Vila de Gracia           3  
3                              Pedralbes           4  
4                            el Poblenou           5  
5                                  Sants           6  
6                       la Vall d'Hebron           7  
7  Vallvidrera, el Tibidabo i les Planes           8

In [82]:
station_info.set_index("station_ID")

Station  Longitude  Latitude  \
station_ID                                                 
1            Barcelona - Ciutadella     2.1874   41.3864   
2              Barcelona - Eixample     2.1538   41.3853   
3                Barcelona - Gràcia     2.1534   41.3987   
4           Barcelona - Palau Reial     2.1151   41.3875   
5              Barcelona - Poblenou     2.2045   41.4039   
6                 Barcelona - Sants     2.1331   41.3788   
7           Barcelona - Vall Hebron     2.1480   41.4261   
8                Barcelona – Observ     2.1211   41.4176   

                                                    Ubication  \
station_ID                                                      
1                                       Parc de la Ciutadella   
2                                  Av. Roma - c/ Comte Urgell   
3           Plaça Gal·la Placídia (Via Augusta - Travesser...   
4                 c/ John Maynard Keynes - c/ de Jordi Girona   
5                 Plaça Josep Trueta (Pujades - Lope de Vega)   
6           Jardins de Can Mantega (Joan Güell - Violant d...   
7           Parc de la Vall d’Hebron - c/ Martí Codolar - ...   
8                                  Ctra Observatori Fabra, 27   

                  District Name                      Neighborhood Name  
station_ID                                                              
1                  Ciutat Vella  Sant Pere, Santa Caterina i la Ribera  
2                      Eixample         la Nova Esquerra de l'Eixample  
3                        Gracia                      la Vila de Gracia  
4                     Les Corts                              Pedralbes  
5                    Sant Marti                            el Poblenou  
6                Sants-Montjuic                                  Sants  
7                Horta-Guinardo                       la Vall d'Hebron  
8           Sarrià-Sant Gervasi  Vallvidrera, el Tibidabo i les Planes

## Data splitting per station

Stations we have in Barcelona: 
1. Barcelona - Ciutadella 
2. Barcelona - Eixample
3. Barcelona - Gràcia 
4. Barcelona - Palau Reial
5. Barcelona - Poblenou
6. Barcelona - Sants
7. Barcelona - Vall Hebron
8. Barcelona - Vallvidrera, el Tibidabo i les Planes

#### BCN- Ciutadella 

In [33]:
BCN_Ciutadella = clean_air_quality[clean_air_quality["Station"]=="Barcelona - Ciutadella"]

BCN_Ciutadella.head()

Empty DataFrame
Columns: [Station, Air Quality, O3 Quality, O3 Value, NO2 Quality, NO2 Value, PM10 Quality, PM10 Value]
Index: []

#### BCN-Eixample

In [34]:
#BCN_Sants = clean_air_quality[clean_air_quality["Station"] == "Barcelona - Sants"]

#BCN_Sants

#BCN_Eixample = 

#clean_air_quality.head()

BCN_Eixample = clean_air_quality[clean_air_quality["Station"]=="Barcelona - Eixample"]

BCN_Eixample.head()

Station Air Quality O3 Quality  O3 Value  \
date_corrected                                                               
2018-01-11 00:00:00  Barcelona - Eixample    Moderate       Good       1.0   
2018-01-11 01:00:00  Barcelona - Eixample        Good       Good       6.0   
2018-01-11 02:00:00  Barcelona - Eixample        Good       Good      14.0   
2018-01-11 03:00:00  Barcelona - Eixample        Good       Good      32.0   
2018-01-11 04:00:00  Barcelona - Eixample        Good       Good      43.0   

                    NO2 Quality  NO2 Value PM10 Quality  PM10 Value  
date_corrected                                                       
2018-01-11 00:00:00    Moderate      113.0         Good        36.0  
2018-01-11 01:00:00        Good       80.0         Good        35.0  
2018-01-11 02:00:00        Good       64.0         Good        34.0  
2018-01-11 03:00:00        Good       44.0         Good        35.0  
2018-01-11 04:00:00        Good       34.0         Good        34.0

In [36]:
station_links = pd.read_csv('air_stations_Nov2017.csv')
station_links

Station  Longitude  Latitude  \
0                             Barcelona - Ciutadella     2.1874   41.3864   
1                               Barcelona - Eixample     2.1538   41.3853   
2                                 Barcelona - Gràcia     2.1534   41.3987   
3                            Barcelona - Palau Reial     2.1151   41.3875   
4                               Barcelona - Poblenou     2.2045   41.4039   
5                                  Barcelona - Sants     2.1331   41.3788   
6                            Barcelona - Vall Hebron     2.1480   41.4261   
7  Barcelona - Vallvidrera, el Tibidabo i les Planes     2.1211   41.4176   

                                           Ubication        District Name  \
0                              Parc de la Ciutadella         Ciutat Vella   
1                         Av. Roma - c/ Comte Urgell             Eixample   
2  Plaça Gal·la Placídia (Via Augusta - Travesser...               Gracia   
3        c/ John Maynard Keynes - c/ de Jordi Girona            Les Corts   
4        Plaça Josep Trueta (Pujades - Lope de Vega)           Sant Marti   
5  Jardins de Can Mantega (Joan Güell - Violant d...       Sants-Montjuic   
6  Parc de la Vall d’Hebron - c/ Martí Codolar - ...       Horta-Guinardo   
7                         Ctra Observatori Fabra, 27  Sarrià-Sant Gervasi   

                       Neighborhood Name  
0  Sant Pere, Santa Caterina i la Ribera  
1         la Nova Esquerra de l'Eixample  
2                      la Vila de Gracia  
3                              Pedralbes  
4                            el Poblenou  
5                                  Sants  
6                       la Vall d'Hebron  
7  Vallvidrera, el Tibidabo i les Planes

## 2. Database connection

In [ ]:
## Import libraries

from sqlalchemy import create_engine


In [ ]:
driver = 'mysql+pymysql:'
user = 'owner'
password = 'ir0n-h4ck'
ip = '104.197.101.244'
database = 'project-week-2-barcelona'


In [ ]:
connection_string = f'{driver}//{user}:{password}@{ip}/{database}'

In [ ]:
engine = create_engine(connection_string)

In [ ]:
#query = """
 #       SELECT * FROM authors
#"""

#query = """
#        CREATE TABLE kerim_table
#        SELECT * FROM authors
#"""
#df_db = pd.read_sql(query, engine)

In [ ]:
#df_db = pd.read_sql(query, engine)

#print(query)


